### DISCUSSION TITLE
How do I verify tokens generated using jose4j library?

### DISCUSSION THREAD

**Discussion by Sumanta9x9** [panva/jose#81] [Q&A]
Here is my code in Java end that generates the token. I need to verify this token from NodeJS

`String salt = "3db71fa3dd384ecd";
Key key = new AesKey(salt.getBytes());
JsonWebEncryption jwe = new JsonWebEncryption();
jwe.setPayload(password);
jwe.setAlgorithmHeaderValue(KeyManagementAlgorithmIdentifiers.A128KW);		jwe.setEncryptionMethodHeaderParameter(ContentEncryptionAlgorithmIdentifiers.AES_128_CBC_HMAC_SHA_256);
jwe.setKey(key);
String serializedJwe = jwe.getCompactSerialization();
System.out.println(serializedJwe);`

**Comment by panva** [comment]
I would appreciate more effort in your question, e.g. format properly and an actual JWE token to test my assumption with. Or even opening the [documentation](https://github.com/panva/jose/blob/v1.26.0/docs/README.md) and seeing there's a JWE module with a decrypt method listed?


```js
const key = jose.JWK.asKey(Buffer.from('3db71fa3dd384ecd'), { alg: 'A128KW' }) // assumes default charset is utf-8 in your system since that's what `salt.getBytes()` uses
const payload = jose.JWE.decrypt(serializedJwe, key, { algorithms: ['A128KW'] })
```
